File: Assets/Game/Script/Framework/Net/CsProtocol/Common/Ping.cs
using Newtonsoft.Json;
using ZJYFrameWork.Net.Core;
using ZJYFrameWork.Net.CsProtocol.Buffer;

namespace ZJYFrameWork.Net.CsProtocol
{
    public class Ping : IPacket
    {
        public static Ping ValueOf()
        {
            var packet = new Ping();
            return packet;
        }

        public short ProtocolId()
        {
            return 103;
        }
    }

    public class PingRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 103;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (packet == null)
            {
                return;
            }

            Ping oPing = (Ping)packet;
            var message = new ServerMessageWrite(oPing.ProtocolId(), oPing);
            var json = JsonConvert.SerializeObject(message);
            Debug.Log(json);
            buffer.WriteString(json);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            Ping packet = new Ping();
            return packet;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/common/ping.lua
---@class Ping
local Ping = class("Ping")
function Ping:ctor()
    
end


---@return Ping
function Ping:new()
    
    return self
end

---@return number
function Ping:protocolId()
    return 103
end

---@return string
function Ping:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return Ping
function Ping:read(data)

    local packet = self:new(
            )
    return packet
end



return Ping
